### DISCUSSION TITLE
In Progress: 3rd Party Components Integration in Blazor Oqtane using IServerStartup and IClientStartup

### DISCUSSION THREAD

**Discussion by ADefWebserver** [oqtane/oqtane.framework#1008] [General] [answered]
I decided to do a Blog post on 

- [IServerStartup](https://github.com/oqtane/oqtane.framework/blob/master/Oqtane.Server/Extensions/ApplicationBuilderExtensions.cs/#L14) and 
- [IClientStartup](https://github.com/oqtane/oqtane.framework/blob/master/Oqtane.Shared/Interfaces/IClientStartup.cs)

Telerik already seems to be covered [at this link](https://github.com/oqtane/oqtane.framework/discussions/690#discussioncomment-55391). So I decided to cover:

- [Radzen](https://razor.radzen.com/)
- [Syncfusion](https://www.syncfusion.com/blazor-components)

However, my main desire is to also implement [MetaWeblog](https://github.com/shawnwildermuth/MetaWeblog) in the Oqtane Blog Module, and it requires a line like this in the **startup.cs**:

`services.AddMetaWeblog<MetaWeblogService>();`

After a lot of searching the web, the only example I could find is a post by @mikecasas (Mike Casas) here: https://github.com/oqtane/oqtane.framework/discussions/693#discussioncomment-53443

It is only doing a normal scoped registration of a service.

@sbwalker can you provide any guidance?

Thanks!

**Reference Links**

- [Startup Services in External Modules](https://github.com/oqtane/oqtane.framework/discussions/693)
- [Register Telerik UI for Blazor](https://github.com/oqtane/oqtane.framework/discussions/690)
- [Trailblazor: Building Dynamic Applications with Blazor (briefly covers IServerStartup)](https://youtu.be/gnYIqYNW54I?t=1731)
- [Load Client Assemblies](https://github.com/oqtane/oqtane.framework/discussions/996)

**Comment by sbwalker** [comment, marked as answer]
@ADefWebserver the IServerStartup interface matches the characteristics of the standard .NET Core Startup methods. You can create a class in your external module's Server project and add whatever service registration logic you would normally add in the Startup class. During startup Oqtane will call the methods automatically.

```
    public class ServerStartup : IServerStartup
    {
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddMetaWeblog<MetaWeblogService>();
        }

        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
        }

        public void ConfigureMvc(IMvcBuilder mvcBuilder)
        {
        }
    }
```

**Reply by ADefWebserver** [reply]
@sbwalker Got it. I hope to have this up in the next two weeks. Thank You

**Reply by Gotiap** [reply]
I tried to get appsettings data into external modules with this IServerStartup, I am getting values into Configure method. But I want to grab some values into ConfigureServices method, So can you please help me for that.

Thank you.

**Reply by ADefWebserver** [reply]
> I tried to get appsettings data into external modules with this IServerStartup, I am getting values into Configure method. But I want to grab some values into ConfigureServices method, So can you please help me for that.
> 
> Thank you.

I have no examples.

**Comment by ADefWebserver** [comment]
I posted a Blog: [Using Radzen In Oqtane Modules](https://blazorhelpwebsite.com/ViewBlogPost/46)

**Comment by ADefWebserver** [comment]
I posted a Blog: [Using Syncfusion In Oqtane Modules](https://blazorhelpwebsite.com/ViewBlogPost/47)

**Reply by Caloke** [reply]
Much obliged!

**Reply by rchibnik** [reply]
This approach worked with Dev express.

**Reply by Ranko-vr** [reply]
It works with MudBlazor too. Thank you for the walkthrough!
